minorminer/utils/raster_embedding.py

```diff
@@ -104,22 +104,29 @@ def find_multiple_embeddings(S, T, timeout=10, max_num_emb=float('inf')):
             from the source to the target in the form of a dictionary with no
             reusue of target variables.
     """
-    _T = T.copy()
     embs = []
-    while True and len(embs) < max_num_emb:
-        # A potential feature enhancement would be to allow different embedding
-        # heuristics here, including those that are not 1:1
-        
+    if inplace:
+        _T = T
         if subgraph_embedding_feasibility_filter(S, T):
             emb = find_subgraph(S, _T, timeout=timeout, triggered_restarts=True)
-        else:
-            emb = []
-        if len(emb) == 0:
-            break
-        else:
-            _T.remove_nodes_from(emb.values())
             embs.append(emb)
-    return embs
+        return embs
+    else:
+        _T = T.copy()
+        while len(embs) < max_num_emb:
```

Comment:
I suggest using a `for` loop instead. for example,

```
# change default for max_num_emb to None
if max_num_emb is None:
    max_num_emb = T.number_of_nodes()
for _ in range(max_num_emb):
    # do stuff
```

reason (subjective): less likely to run into an infinite loop with a for loop of fixed number of iterations.

Reply:
```max_num_emb = int(T.number_of_nodes()/S.number_of_nodes())```  # to be more elegant

Reply:
I would be inclined to include a statement like this, i.e. ignore inplace if max_num_embeddings=1
if max_num_embeddings == 1:
    inplace = True

Reply:
By statement do you mean comments?